pkgs/applications/editors/emacs/generic.nix

```diff
@@ -67,7 +67,31 @@ let emacs = (if withMacport then llvmPackages_6.stdenv else stdenv).mkDerivation
   pname = pname + lib.optionalString ( !withX && !withNS && !withMacport && !withGTK2 && !withGTK3 ) "-nox";
   inherit version;
 
-  patches = patches fetchpatch;
+  patches = patches fetchpatch ++
+            # TODO: This patch is written awkwardly because at the
+            # time of writing, the defcustom for
+            # native-comp-driver-options is set to a different value
+            # in Emacs 28.2 as compared to Emacs 28-unstable and Emacs
+            # 29.  Upon removing Emacs 28 from Nixpkgs, it is
+            # recommended that we patch the defcustom directly,
+            # instead of finding a suitable place afterward to setq.
```

Comment:
The point of the patch is to break when upstream changes something. This way there is no point in using a patch.

If we're going to do patches, we need different patches for different versions.